### PR TITLE
Feature: Leo function inline and visibility

### PR DESCRIPTION
**Pull request by collinc97**
Adds documentation for inline functions and visibility in Leo.

### CODE REVIEW

File: documentation/leo/03_language.md [hunk 1]
```diff
@@ -194,9 +194,13 @@ struct array3 {
...
 A [record](../concepts/02_records.md) data type is declared as `record {name} {}`.
-Records contain component declarations `{name}: {type},`.
+Records contain component declarations `{visibility} {name}: {type},`.
+
+A visibility can be either `constant`, `public`, or `private`.
+Users may also omit the visibility, in which case, Leo will default to `private`.
+
 Record data structures must contain the `owner` and `gates` components as shown below.
-When passing a record as input to a program function, the `_nonce: group,` component is also required
+When passing a record as input to a program function, the `_nonce: group`, component is also required
```
Comment (acoglio) [inline documentation/leo/03_language.md:203]: `_nonce: group`, => `_nonce: group` — i.e. no comma

File: documentation/leo/03_language.md [hunk 2]
```diff
@@ -267,8 +271,10 @@ transition foo(public a: field) -> field {
 ### Helper Function
 
 A helper function is declared as `function {name}() {}`.
-Helper functions contain expressions and statements that can compute values.
-Helper functions that cannot be executed directly. Helper functions must be called by other functions.
+Helper functions contain expressions and statements that can compute values,
+however helper functions cannot produce `records`.
```
Comment (acoglio) [inline documentation/leo/03_language.md:275]: Can they take records? If they don't, we might say 'cannot consume or produce'. If they do, the text is good as is.

File: documentation/leo/03_language.md [hunk 3]
```diff
@@ -267,8 +271,10 @@ transition foo(public a: field) -> field {
...
+
+Helper functions that cannot be executed directly. Instead, they must be called by other functions.
```
Comment (acoglio) [inline documentation/leo/03_language.md:277]: There's a stray 'that'.

You may also consider rephrasing the first sentence as "Helper functions cannot be called directly from the outside." (Not sure which one is better though.)

File: documentation/leo/03_language.md [hunk 4]
```diff
@@ -281,6 +287,24 @@ function foo(
 }
 ```
 
+### Inline Function
+
+An inline function is declared as `inline {name}() {}`.
+Inline functions contain expressions and statements that can compute values.
+Inline functions that cannot be executed directly,
```
Comment (acoglio) [inline documentation/leo/03_language.md:294]: Remove stray 'that'. Also, you could consider rephrasing (or not) as proposed above for helper functions.

File: documentation/leo/03_language.md [hunk 5]
```diff
@@ -281,6 +287,24 @@ function foo(
 }
 ```
 
+### Inline Function
+
+An inline function is declared as `inline {name}() {}`.
+Inline functions contain expressions and statements that can compute values.
+Inline functions that cannot be executed directly, 
+instead the Leo compiler “inlines” the body of the function at each call site.
```
Comment (acoglio) [inline documentation/leo/03_language.md:295]: You could omit the double quotes around "inlines": it's a proper technical term.

**Review by acoglio — APPROVED** [review]
Looks good, just some minor comments.

File: documentation/leo/03_language.md (hunk 1 above)
Comment (collinc97) [inline documentation/leo/03_language.md:203]:
```suggestion
When passing a record as input to a program function, the `_nonce: group` component is also required
```

File: documentation/leo/03_language.md (hunk 3 above)
Comment (collinc97) [inline documentation/leo/03_language.md:277]:
```suggestion
Helper functions cannot be called directly from the outside. Instead, they must be called by other functions.
```

File: documentation/leo/03_language.md (hunk 4 above)
Comment (collinc97) [inline documentation/leo/03_language.md:294]:
```suggestion
Inline functions cannot be executed directly from the outside, 
```

File: documentation/leo/03_language.md (hunk 5 above)
Comment (collinc97) [inline documentation/leo/03_language.md:295]:
```suggestion
instead the Leo compiler inlines the body of the function at each call site.
```

**Review by collinc97 — COMMENTED** [review]
Address Feedback

**Pull request merged**